symengine/complex.h

```diff
@@ -64,15 +64,14 @@ class Complex : public Number {
     static RCP<const Number> from_two_nums(const Number &re,
         const Number &im);
 
+    //! \return `true` if exactly `0`
+    virtual bool is_exact_zero() const { return false; }
     //! \return `true` if `0`
-    virtual bool is_zero() const { return ((this->real_ == 0) && (this->imaginary_ == 0)); }
+    virtual bool is_zero() const { return false; }
```

Comment:
If imaginary part is zero we return a rational or an integer. These checks are only valid for Complex numbers that are not in canonical form